internal/services/controller/controller.go

```diff
@@ -267,6 +257,44 @@ func (c *Controller) Run(ctx context.Context) error {
 	return nil
 }
 
+func (c *Controller) startConditionalInformersWithWatcher(ctx context.Context, conditionalInformers []conditionalInformer) {
+	if err := wait.PollImmediateInfiniteWithContext(ctx, time.Minute*2, func(ctx context.Context) (done bool, err error) {
+		if !isClusterAPIAvailable(c.discovery, c.log) {
+			return false, nil
+		}
+
+		c.log.Infof("Cluster API server is available, starting conditional informers")
+		for _, informer := range conditionalInformers {
+			if isResourceAvailable(c.discovery, c.log, informer.groupVersion, informer.apiType) {
```

Comment:
the isResourceAvailable fetches the resource list every time it's called, maybe we should avoid that?